### ISSUE TITLE
Model Query Filter generates incorrect SQL

### ISSUE THREAD

**Issue by bradmarder** [aspnet/EntityFrameworkCore#10785]
The combination of a query filter on a model and a nullable property on a navigation property will generate incorrect SQL. Please note the missing `OR[x.Blog].[IsNullable] IS NULL` in the first query.

Bypassing the model query filter with `.IgnoreQueryFilters()` will generate the correct SQL.

    public class Blog
    {
        public Guid Id { get; set; }
        public bool IsDeleted { get; set; }
        public bool? IsNullable { get; set; }
    }
    public class Post
    {
        public Guid Id { get; set; }
        public Guid BlogId { get; set; }
        public virtual Blog Blog { get; set; }
    }
    public class TestContext: DbContext
    {
        public DbSet<Blog> Blogs { get; set; }
        public DbSet<Post> Posts { get; set; }

        protected override void OnModelCreating(ModelBuilder builder)
        {
            builder.Entity<Blog>().HasQueryFilter(x => !x.IsDeleted);
        }
    }

`await db.Posts.Where(x => x.Blog.IsNullable != true).CountAsync();`

```
SELECT COUNT(*)
FROM[Posts] AS[x]
INNER JOIN(
    SELECT[b].*
    FROM[Blogs] AS[b]
    WHERE[b].[IsDeleted] = 0
) AS[t] ON[x].[BlogId] = [t].[Id]
WHERE[t].[IsNullable] <> 1
```

   `await db.Posts.Where(x => x.Blog.IsNullable != true).IgnoreQueryFilters().CountAsync();`

```
SELECT COUNT(*)
FROM[Posts] AS[x]
INNER JOIN[Blogs] AS[x.Blog] ON[x].[BlogId] = [x.Blog].[Id]
WHERE([x.Blog].[IsNullable] <> 1) OR[x.Blog].[IsNullable] IS NULL
```

    
 

### Further technical details
EF Core version: Microsoft.AspNetCore.All 2.05
Database Provider: Microsoft.EntityFrameworkCore.SqlServer
Operating system: win 10  64-bit
IDE: (e.g. Visual Studio 2017 15.4)

**Comment by smitpatel** [comment]
https://github.com/aspnet/EntityFrameworkCore/blob/7a54e8cf63c5f2e8206c53443f00d8d06ce64ae1/src/EFCore.Relational/Query/ExpressionVisitors/Internal/IsNullExpressionBuildingVisitor.cs#L80-L103

Above method should also evaluate ColumnReferenceExpression to check nullability. The bug is not related to QueryFilter but rather due to nullable column being selected from subquery.

**Comment by maumar** [comment]
fixed in 154725799c44ac14e7592c17c2655037fcf76261
